storage/innobase/fil/fil0fil.cc

```diff
@@ -7948,33 +7948,42 @@ fil_set_encryption(
 
 /** Enable encryption of temporary tablespace
 @param[in,out]	space	tablespace object
+@param[in]	enable	true to enable encryption, false to disable
 @return DB_SUCCESS on success, DB_ERROR on failure */
 dberr_t
 fil_temp_update_encryption(
-	fil_space_t*	space)
+	fil_space_t*	space,
+	bool		enable)
 {
-	/* Make sure the keyring is loaded. */
-	if (!Encryption::check_keyring()) {
-		ib::error() << "Can't set temporary tablespace"
-			<< " to be encrypted because"
-			<< " keyring plugin is not"
-			<< " available.";
-			return(DB_ERROR);
-	}
+	if (enable && !FSP_FLAGS_GET_ENCRYPTION(space->flags)) {
```

Comment:
I would rewrite it as if (!enable) {return (DB_SUCCESS) } at top. and keep the rest same. No indentation change required and fewer lines changed.

Reply:
done and force pushed